File: cis/lib/fetchdata/izinkeluar.dart
class IzinKeluar {
  final int id;
  final int mahasiswaId;
  final String rencanaBerangkat;
  final String rencanaKembali;
  final String keperluan;
  final int status;

  IzinKeluar({
    required this.id,
    required this.mahasiswaId,
    required this.rencanaBerangkat,
    required this.rencanaKembali,
    required this.keperluan,
    required this.status,
  });

  factory IzinKeluar.fromJson(Map<String, dynamic> json) {
    return IzinKeluar(
      id: json['id'],
      mahasiswaId: json['mahasiswa_id'],
      rencanaBerangkat: json['RencanaBerangkat'],
      rencanaKembali: json['RencanaKembali'],
      keperluan: json['Keperluan'],
      status: json['status'],
    );
  }
}

File: web-service/app/Models/IzinKeluar.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Notifications\Notifiable;

class IzinKeluar extends Model
{
    use HasFactory, Notifiable;
    protected $fillable = [
        'mahasiswa_id',
        'RencanaBerangkat',
        'RencanaKembali',
        'Keperluan',
        'status',
    ];

    public function mahasiswa()
    {
        return $this->belongsTo(Mahasiswa::class, 'mahasiswa_id');
    }
}
